props/saber_fett263_buttons.h

```diff
@@ -293,248 +662,3382 @@ class SaberFett263Buttons : public virtual PropBase {
+#endif
+    TRACE(PROP, "end");
+  }
+
+// Update Font / Save Style in Edit Mode, skips Preon effect (except for Preon Editing previews) using FastOn
+  virtual void UpdateFont(int preset_num) {
+    TRACE(PROP, "start");
+    bool on = SaberBase::IsOn();
+    if (on) Off();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+#define UNSET_BLADE_STYLE(N) \
+    delete current_config->blade##N->UnSetStyle();
+    ONCEPERBLADE(UNSET_BLADE_STYLE)
+    current_preset_.SetPreset(preset_num);
+#define SET_BLADE_STYLE(N) \
+    current_config->blade##N->SetStyle(style_parser.Parse(current_preset_.current_style##N.get()));
+    ONCEPERBLADE(SET_BLADE_STYLE)
+    chdir(current_preset_.font.get());
+#ifdef SAVE_COLOR_CHANGE
+    SaberBase::SetVariation(current_preset_.variation);
+#else
+    SaberBase::SetVariation(0);
+#endif
+    if (on) {
+#ifdef FETT263_EDIT_MODE_MENU
+      if (menu_type_ == MENU_PREON_OPTION || menu_type_ == MENU_PREON_SIZE) {
+        On();
+      } else {
+        FastOn();
+      }
+#else
+      FastOn();
+#endif      
+    }
+    TRACE(PROP, "end");
+  }
+
+  bool Parse(const char *cmd, const char* arg) override {
+    if (PropBase::Parse(cmd, arg)) return true;
+    if (!strcmp(cmd, "list_current_tracks")) {
+      // Tracks must be in: font/tracks/*.wav 
+      LOCK_SD(true);
+      for (const char* dir = current_directory; dir; dir = next_current_directory(dir)) {
+        PathHelper path(dir, "tracks");
+        ListTracks(path);
+      }
+      LOCK_SD(false);
+      return true;
+    }
+    return false;
+  }
+
+  // Go to first Preset.
+  virtual void first_preset() {
+#ifdef SAVE_PRESET
+    SaveState(0);
+#endif
+    UpdateFont(0);
+}
+
+  // Go to the next Preset skipping Preon effect with FastOn.
+  virtual void next_preset_fast() {
+#ifdef SAVE_PRESET
+    SaveState(current_preset_.preset_num + 1);
+#endif
+    UpdateFont(current_preset_.preset_num + 1);
+  }
+
+  // Go to the previous Preset skipping Preon effect with FastOn.
+  virtual void previous_preset_fast() {
+#ifdef SAVE_PRESET
+    SaveState(current_preset_.preset_num - 1);
+#endif
+    UpdateFont(current_preset_.preset_num - 1);
+  }
+
+#ifdef FETT263_DUAL_MODE_SOUND
+  // Select out.wav based on blade orientation, up = odd, down = even
+  void SelectIgnitionSound() {
+      Effect* effect;
+      if (SFX_poweron) {
+        effect = &SFX_poweron;
+      } else {
+        effect = &SFX_out;
+      }
+      int f = effect->files_found();
+      if (fusor.angle1() > 0) {
+        f = (rand() % ((f + 1)/ 2)) * 2;
+      } else {
+        f = 1 + (rand() % (f / 2)) * 2;
+      }
+      effect->Select(f);
+    }
+#endif
+
+  void PlayQuote() {
+    if (SFX_quote) {
+      SFX_quote.SelectNext();
+      wav_player->PlayOnce(&SFX_quote, 0.0);
+    } else {
+      SFX_force.SelectNext();
+      wav_player->PlayOnce(&SFX_force, 0.0);
+    }
+  }
+
+  // Fast On Gesture Ignition
```

Comment:
We need to discuss how to write better comments.
This comment is not very good. The first part just repeats the name of the function, and the second two words talks about gestures, but that's confusing because this function doesn't do anything with gestures.

I think the comment should be something like this:

// FastOn(), like On(), but skips the preon stage so that ignition happens immediately.
// Primarily intended to be used with gestures where an immediate response makes more sense.

This describes what the function does, what it's for and how it's intended to be used.
With a description like this, callers don't need to read the code to understand it.

Reply:
I think this is an old comment from OS5, but I've updated will try to be more descriptive in comments in future